src/video_core/renderer_vulkan/vk_graphics_pipeline.cpp

```diff
@@ -286,28 +287,69 @@ void GraphicsPipeline::BindResources(Core::MemoryManager* memory, StreamBuffer&
         return offset;
     };
 
-    std::array<vk::Buffer, MaxVertexBufferCount> buffers;
-    std::array<vk::DeviceSize, MaxVertexBufferCount> offsets;
-    VAddr base_address = 0;
-    u32 start_offset = 0;
+    const auto cmdbuf = scheduler.CommandBuffer();
 
-    // Bind vertex buffer.
     const auto& vs_info = stages[0];
-    const size_t num_buffers = vs_info.vs_inputs.size();
-    for (u32 i = 0; i < num_buffers; ++i) {
-        const auto& input = vs_info.vs_inputs[i];
-        const auto buffer = vs_info.ReadUd<AmdGpu::Buffer>(input.sgpr_base, input.dword_offset);
-        if (i == 0) {
-            start_offset = map_staging(buffer.base_address.Value(), buffer.GetSize());
-            base_address = buffer.base_address;
+    if (!vs_info.vs_inputs.empty()) {
```

Comment:
This has grown quite big, I think it will be easier on the eyes when separated into a member function